tests/Integration/Behaviour/Features/Context/Domain/Product/UpdateAttachmentFeatureContext.php

```diff
@@ -61,7 +62,9 @@ public function associateProductAttachment(string $attachmentReference, string $
      */
     public function assertProductAttachments(string $productReference, array $attachmentReferences): void
     {
-        $attachmentIds = $this->getProductForEditing($productReference)->getAssociatedAttachmentIds();
+        $attachmentIds = array_map(function (ProductAttachmentInfo $attachment): int {
```

Comment:
This means you only check the returned IDs Since now more data are returned All of them should be tested